src/main/java/gregtech/client/event/FluidVisualHandler.java

```diff
@@ -0,0 +1,217 @@
+package gregtech.client.event;
+
+import gregtech.api.GTValues;
+import gregtech.api.fluids.MaterialFluidBlock;
+import gregtech.api.unification.material.Material;
+import net.minecraft.block.Block;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.BufferBuilder;
+import net.minecraft.client.renderer.EntityRenderer;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.init.MobEffects;
+import net.minecraft.potion.PotionEffect;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.client.event.EntityViewRenderEvent;
+import net.minecraftforge.client.event.RenderBlockOverlayEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Handles the rendering when a player is submerged in a GT fluid block
+ *
+ * <p>
+ * Changes the submerged fog to the color of the fluid
+ * Changes the fluid overlay texture to the color of the fluid
+ * Fixes FOV scaling if fluids change player speed
```

Comment:
This comment is not quite correct. It will scale the FOV regardless of if movement speed is changed, as that is not checked.